### ISSUE TITLE
[GCP] enable GPUDirect-TCPX for H100s

### ISSUE THREAD

**Issue by asaiacai** [skypilot-org/skypilot#3592]
The default skypilot configuration doesn't take advantage of some the high bandwidth networking available on GCP H100 instances. Right now if I benchmark NCCL on 2xH100:8 I only get 15% of the advertised bandwidth for these instances (151Gbps out of 1000Gbps).  We'd need to enable [GPUDirect-TCPx](https://cloud.google.com/compute/docs/gpus/gpudirect) to accomplish this. One user already requested this.

This was my all reduce busbw as measured by following this [torch all-reduce bandwidth measurement.](https://github.com/stas00/ml-engineering/blob/master/network/benchmarks/all_reduce_bench.py)
```
(head, rank=0, pid=22990) [sky-1aa7-asai-ebd1-head-9nftmsu6-compute:0]:The average bandwidth of all_reduce with a 40.0GB payload (5 trials, 16 ranks):
(head, rank=0, pid=22990) [sky-1aa7-asai-ebd1-head-9nftmsu6-compute:0]: algbw: 11.396 GBps (91.2 Gbps)
(head, rank=0, pid=22828) [sky-1aa7-asai-ebd1-head-9nftmsu6-compute:0]: busbw: 18.973 GBps (151.8 Gbps)
```

**Comment by jiangwei221** [comment]
It seems that enabling GPUDirect-TCPX requires Container optimimzed OS, and this os locked down, and can't install packages in the VM OS level.

**Comment by Michaelvll** [comment]
This is now resolved by #5553
